src/node/internal/internal_http_util.ts

```diff
@@ -24,3 +24,30 @@ export function once<RT>(
 
 export const kServerResponse = Symbol('ServerResponse');
 export const kIncomingMessage = Symbol('IncomingMessage');
+
+// RFC 7230 compliant header value splitting that respects quoted-string constructions
+// Spec: https://www.rfc-editor.org/rfc/rfc7230.html#appendix-B
+// Edge cases handled:
+// - Quoted strings with commas: 'text/plain; param="a, b"' -> 'text/plain; param="a, b"'
+// - Escaped quotes in strings: 'text/plain; param="a \"quoted\" value"' -> 'text/plain; param="a \"quoted\" value"'
+// - Multiple values: 'text/plain; f="a, b", text/html' -> 'text/plain; f="a, b"'
+// - Whitespace before commas: 'value1 \t , value2' -> 'value1'
+// - No commas: 'single-value' -> 'single-value'
+export function splitHeaderValue(value: string): string {
+  let inQuotes = false;
+
+  for (let i = 0; i < value.length; i++) {
+    const char = value[i];
+
+    if (char === '"') {
+      inQuotes = !inQuotes;
+    } else if (char === '\\' && inQuotes) {
+      i++; // Skip next character (it's escaped)
+    } else if (!inQuotes && char === ',') {
+      // Found unquoted comma, trim whitespace and return
+      return value.slice(0, i).trimEnd();
```

Comment:
This, and the might need to trim the start as well. If you look at the definition of a header field in the spec, the `field-value` may have OWS before and after.

```
header-field = field-name ":" OWS field-value OWS
```

Likewise, the `return value` below may need to be trimmed.

For example, the following is a valid header field:

```
foo:			abc			,			xyz			,			123
```

In which case the values must be [`abc', 'xyz', '123']

Reply:
that said, if these are coming from the `Headers` class we can be reasonably certain that the leading whitespace is removed but having a test that verifies would be helpful.

Reply:
> ... I think you're mixing places where we need Headers class.

I'm not following. My comment was about whether we're certain that we don't need to trim leading whitespace from the value. I *think* the `Headers` class here provides properly trimmed values but I just want to be certain.